docs/PATCH-ENVELOPES.md

```diff
@@ -1,36 +1,99 @@
 # Patch Envelopes
 
-## Overview
+## What is Patch Envelope? 
 
-Patch Envelopes are objects which are exposed to app instances via EVE meta-data server.
-This objects can be useful to update any kind of information in a secure, isolated manner
-(for instance, configuration parameters) on app instance without the need of rebooting it.
-Alternatively, one can create new image for app instance, upload it to EVE, purge and restart
-application instance with the new image, even if there was a small change in some configuration
-file. To top that there will be a down-time during reboot of app instance. To summarise it,
-patch envelopes goal is to make fleet management of app instances easier.
+Patch Envelope is a functionality in EVE that helps you change data in your App Instance in
+runtime without the need to reboot instance, or recreate image. Imagine that in order to change
+configuration in one of your program, you have to create new OS image and flush it to your computer.
+With Patch Envelope you don’t have to do that. Patch Envelope constitutes of Binary Artefacts (blobs)
+which are base64-encoded objects this can be anything, from binary file to a configuration yaml file.
+Those Patch Envelopes are exposed to App Instance via Metadata server.
 
-## Patch Envelope structure
+## Why should I use it?
 
-Patch Envelope are created on controller and propagated to EVE via protobuf's `EvePatchEnvelope` message.
-It consists of:
+When recreating image doesn’t make sense (i.e. you just want to change one configuration parameter and
+you don’t want to recreate Windows VM) or when downtime is not an option for you
```

Comment:
don’t want to recreate Windows VM -> don’t want to recreate VM image
(let's not make this Windows specific)